File: yujin-level0/Week61/같은 숫자는 싫어.js
function solution(arr) {
  var answer = [arr[0]];

  for (let i = 1; i < arr.length; i++) {
    if (answer[answer.length - 1] !== arr[i]) {
      answer.push(arr[i]);
    }
  }

  return answer;
}

// 다른 풀이
// function solution(arr) {
//   return arr.filter((v, i) => v !== arr[i - 1]);
// }

File: gnsals0904/level1/같은_숫자는_싫어.java
import java.util.*;

public class Solution {
    public int[] solution(int []arr) {
        int prev = -1;
        ArrayList<Integer> answers = new ArrayList<>(1_000_000);
        for(int a : arr){
            if(prev != a) answers.add(a);
            prev = a;
        }
        return answers.stream()
                .mapToInt(Integer::intValue)
                .toArray();
    }
}